stages/node-fast.yaml

```diff
@@ -48,7 +48,7 @@ spec:
       {{ with .status.addresses }}
       {{ YAML . 1 }}
       {{ else }}
-      {{ with NodeIP }}
+      {{ with NodeIP .metadata.name }}
```

Comment:
I have to make it the same because it's sharing the same FuncMap variable. 
Otherwise, I got the following error in the unit test.
```
{"time":"2023-02-20T19:48:15.204406+09:00","level":"ERROR","source":"/Users/aa404681/Documents/internal_ws/cni/kwok/pkg/kwok/controllers/pod_controller.go:281","msg":"Failed to configure pod","pod":"default/pod-with-hostNetwork","node":"node0","err":"template: _:47:5: executing \"_\" at \u003cNodeIP\u003e: wrong number of args for NodeIP: want 0 got 1"}
```
It's the reason I changed the key in pod controller to `NodeIPMap`.

Reply:
I have to make it the same because it's sharing the same FuncMap variable. 
Otherwise, I got the following error in the unit test.
```
{"time":"2023-02-20T19:48:15.204406+09:00","level":"ERROR","source":"_/cni/kwok/pkg/kwok/controllers/pod_controller.go:281","msg":"Failed to configure pod","pod":"default/pod-with-hostNetwork","node":"node0","err":"template: _:47:5: executing \"_\" at \u003cNodeIP\u003e: wrong number of args for NodeIP: want 0 got 1"}
```
It's the reason I changed the key in pod controller to `NodeIPMap`.

Reply:
I think it is the problem of unit test. funcMap are shared but the NewNodeController and NewPodController are called in a random order (unlike in `controller.go`) . The above error occurs because NewPodController is called before NewNodeController.

Reply:
It seems to be a rather tangled place, do you have any good ideas?

Reply:
I think we should separate NodeIP and NodeIPMap because there is no need to check the NodeInfo map in the node controller. Different function should have different key. If we share the same key name, it will be overridden anyway. 

Another idea I have is,  instead of using a shared map, making a new function map when newRenderer for each controller.

Reply:
Another comment is about the unit test. Maybe we should also consider using test suite like ginkgo and define all controller before suite in the same way that we have in the controller.go.

Reply:
> Another idea I have is, instead of using a shared map, making a new function map when newRenderer for each controller.

I am also more inclined to this approach

> Another comment is about the unit test. Maybe we should also consider using test suite like ginkgo and define all controller before suite in the same way that we have in the controller.go.

I agree with you, but this transition should not be done in this PR, we can start a new PR for this work.

`ginkgo` is usually used to write e2e or integration tests, currently e2e is written in shell, maybe we can use it to fill in the integration tests part.

Reply:
Made the changes by re-defining `commonFuncMap`.
https://github.com/kubernetes-sigs/kwok/compare/d484e263f4a8182cc3e79ae29117827a88d32aa0..17fe8647d6db8e631d96314d8359441948b74001

Reply:
I am not sure what is the purpose of shared funcMap (which is changed to commonFuncMap). 
Do you want it to be a global variable that always being added (common) or it is just a default variable for config value?  
For the former case, the `conf.FuncMap` should be controller-specific (i.e., not passing commonFuncMap) and copying commonFuncMap should be there. 
For the latter case, copying commonFuncMap (funcMap) is not needed. 

I will commit a new amend change regarding the purpose.

Reply:
https://github.com/kubernetes-sigs/kwok/blob/8e2871ed268e8efbd3934eeb5432a9afe6844ac2/pkg/kwok/controllers/node_controller.go#L121

https://github.com/kubernetes-sigs/kwok/blob/8e2871ed268e8efbd3934eeb5432a9afe6844ac2/pkg/kwok/controllers/pod_controller.go#L125

This seems to be an omission from my previous Review where the global variable was changed, which is surprising, let's fix it.

```
funcMap := template.FuncMap{ 
```

Reply:
Made changes (I suppose you prefer the latter case. So, I renamed it to defaultFuncMap).
https://github.com/kubernetes-sigs/kwok/compare/17fe8647d6db8e631d96314d8359441948b74001..8943c6bcf4f05b5fba5f833ea0bb9905d1783a57